src/particles/spacecharge/GatherAndPush.cpp

```diff
@@ -111,11 +129,56 @@ namespace impactx::particles::spacecharge
                         px += field_interp[0] * push_consts * dr[2] / (beta * c0_SI);
                         py += field_interp[1] * push_consts * dr[2] / (beta * c0_SI);
                         pz += 0.0_rt;
-                        //pz += field_interp[2] * push_consts;  // TODO: non-zero in 2.5D, but we will add a toggle to turn it off there, too
 
                         // push position is done in the lattice elements
                     });
                 }
+                if (space_charge == SpaceChargeAlgo::True_2p5D) {
+                    // flatten 3rd dimension
+                    auto prob_lo_2D = gm.ProbLoArray();
+                    prob_lo_2D[2] = 0.0_rt;
+
+                    // group together constants for the momentum push
+                    amrex::ParticleReal const chargesign = charge / std::abs(charge);
+
+                    amrex::ParallelFor(np, [=] AMREX_GPU_DEVICE (int i) {
+                        // access SoA Real data
+                        amrex::ParticleReal & AMREX_RESTRICT x = part_x[i];
+                        amrex::ParticleReal & AMREX_RESTRICT y = part_y[i];
+                        amrex::ParticleReal z = 0.0_prt;  // flatten 3rd dimension
+                        amrex::ParticleReal & AMREX_RESTRICT px = part_px[i];
+                        amrex::ParticleReal & AMREX_RESTRICT py = part_py[i];
+                        amrex::ParticleReal & AMREX_RESTRICT pz = part_pz[i];
+
+                        // force gather
+                        amrex::GpuArray<amrex::Real, 3> const field_interp =
+                            ablastr::particles::doGatherVectorFieldNodal<2>(
+                                x, y, z,
+                                scf_arr_x, scf_arr_y, scf_arr_z,
+                                invdr,
+                                prob_lo_2D
+                            );
```

Comment:
After passing also `m_phi` by reference into `GatherAndPush (...)`:
```suggestion
                            ablastr::particles::doGatherScalarFieldNodal<2>(
                                x, y, z,
                                phi_arr,
                                invdr,
                                prob_lo_2D
                            );
```